### ISSUE TITLE
The expression after '&' in a pipeline element produced an object that was not valid. It must result in a command name, a script block, or a CommandInfo object.

### ISSUE THREAD

**Issue by daggerok** [ScoopInstaller/Scoop#3845]
Hello,

How I can fix this?

```batch
C:\>scoop update
Updating Scoop...
The expression after '&' in a pipeline element produced an object that was not valid. It must result in a command
name, a script block, or a CommandInfo object.
At C:\local\scoop\shims\scoop.ps1:2 char:71
+ ... on.expectingInput) { $input | & $path  @args } else { & $path  @args  ...
+                                                             ~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : BadExpression

WARN  Overwriting shim to scoop.ps1 installed from
Updating 'extras' bucket...
```

Thanks

---
Maksim

**Comment by nickosh** [comment]
Same problem with any scoop command after scoop update.

**Comment by John-Barr** [comment]
I have the same problem but the error is coming from git:

```
The expression after '&' in a pipeline element produced an object that was not valid. It must result in a 
command name, a script block, or a CommandInfo object.
At C:\Scoop\Applications\apps\scoop\current\lib\git.ps1:7 char:7
+     & "$env:COMSPEC" /c $cmd
+       ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:String) [], RuntimeException
    + FullyQualifiedErrorId : BadExpression
```

Surely this is because of some environment variable problem, right?

**Comment by h404bi** [comment]
@John-Barr Please check and post the `$env:COMSPEC` env of your PC.

**Comment by John-Barr** [comment]
@h404bi via CMD
```
> echo %COMSPEC%
C:\WINDOWS\SYSTEM32\cmd.exe
```

**Comment by John-Barr** [comment]
After scratching my head for days, I thought to simply check the env variables from the Windows GUI itself and sure enough, my `%COMSPEC%` variable was missing.

CMD was able to return the correct path even though it did not exist... I do not know how this has happened but I hope @daggerok's solution is as simple as mine was. 

Apologies @h404bi if any time was wasted on such a silly issue. Cheers.

**Comment by nickosh** [comment]
> @nickosh Please post the error output.

Sorry, I'm already reinstalled Scoop and can't  reproduct this error anymore. In my case I deleted scoop from apps folder and run install script. So far its works. If I catch this error again, I post it here. I remember it happens after upgrading one app, but dont remember which.

**Comment by mitranim** [comment]
Just had a similar issue. My `COMSPEC` was missing because I deleted it manually a while ago due to some weird issues, and I had to reinstate it. Perhaps `scoop` should fall back on `cmd` when this env var is missing / empty.

**Comment by rashil2000** [comment]
Closed in #4692
